src/main/play_json_2.4/com/codacy/client/bitbucket/client/BitbucketClient.scala

```diff
@@ -4,8 +4,5 @@ import com.codacy.client.bitbucket.WSWrapper
 import com.codacy.client.bitbucket.WSWrapper.WSClient
 import com.codacy.client.bitbucket.client.Authentication.Credentials
 
-import scala.util.Try
-
-class BitbucketClient(credentials: Credentials) extends BitbucketClientBase(credentials) {
-  override protected def buildClient(): WSClient = WSWrapper.build()
-}
+class BitbucketClient(client: WSClient = WSWrapper.build(), credentials: Credentials)
```

Comment:
to be on the safe side and to keep going with the original semantics, I would take as argument a function e.g.:

```scala
() => WSClient
```

keep the `WithClient` method:

```scala
  private var client: WSClient = null
  private def withClient[T](block: WSClient => T): Try[T] = {
    try {
      if (client == null) {
        client = buildClient()        
     }

     Try(block(client))
    } finally {
     if (client != null) client.close()
     client = null
    }
  }
```